File: Scala/scalabasic/src/main/java/TestVar.java
import java.util.ArrayList;

/**
 * @Author Jiaman
 * @Date 2021/5/6 9:40
 * @Desc
 */

public class TestVar {
    // 全局变量
    String name1 = "feishou";
    public static void main(String[] args) {
        // 内部变量
        String name = "zhangq";
        int age = 19;
        System.out.println("name=" + name + "age=" +age);

        // 变量修改
        name = "xyz";
        System.out.println(name);


    }
}

File: Scala/scalabasic/src/main/scala/fun/shijin/scalabasic/chapter01/TestVar.scala
package fun.shijin.scalabasic.chapter01

/**
 * @Author shijin
 * @Date 2021/5/6 9:44
 * @Desc
 */

object TestVar {
  // 声明外部变量
  val name1:String = "java"
  val age1:Int = 19
  def main(args: Array[String]): Unit = {
  
    //声明内部变量 ， val 不可修改，var可修改
    var name: String = "zhangq"
    val age: Int = 19
  
    println(s"name=$name,age=$age")
  
    name = "python"
    println(name)
    
    // 声明age为Double类型
    val age2:Double = 19
    println(age2)
    
    // 不指定类型，自动推断类型
    var age3 = 19
    println(age3)
    
    val name4, age4 = 10
    
    println(name4 + age4)
    
  }
}
